plugins/module_utils/openshift_adm_prune_auth.py

```diff
@@ -0,0 +1,339 @@
+
+        if len(result) == 0:
+            return [], False
+
+        def _update_user_group(binding_namespace, subjects):
+            users, groups = [], []
+            for x in subjects:
+                if x['kind'] == 'User':
+                    users.append(x['name'])
+                elif x['kind'] == 'Group':
+                    groups.append(x['name'])
+                elif x['kind'] == 'ServiceAccount':
+                    namespace = binding_namespace
+                    if x.get('namespace') is not None:
+                        namespace = x.get('namespace')
+                    if namespace is not None:
+                        users.append("system:serviceaccount:%s:%s" % (namespace, x['name']))
+            return users, groups
+
+        candidates = []
+        changed = False
+        for item in result:
+            subjects = item.get('subjects', [])
+            retainedSubjects = [x for x in subjects if x['kind'] == ref_kind and x['name'] in ref_names]
+            if len(subjects) != len(retainedSubjects):
+                updated_binding = item
+                updated_binding['subjects'] = retainedSubjects
+                binding_namespace = item['metadata'].get('namespace', None)
+                updated_binding['userNames'], updated_binding['groupNames'] = _update_user_group(binding_namespace, retainedSubjects)
+                candidates.append(binding_namespace + "/" + item['metadata']['name'] if binding_namespace else item['metadata']['name'])
+                if not self.check_mode:
+                    try:
+                        resource.apply(updated_binding, namespace=binding_namespace)
+                        changed = True
+                    except DynamicApiError as exc:
+                        msg = "Failed to apply object due to: {0}".format(exc.body)
+                        self.fail_json(msg=msg)
+        return candidates, changed
+
+    def update_security_context(self, ref_names, key):
+        params = {'kind': 'SecurityContextConstraints', 'api_version': 'security.openshift.io/v1'}
+        sccs = self.kubernetes_facts(**params)
+        if not sccs['api_found']:
+            self.fail_json(msg=sccs['msg'])
+        sccs = sccs.get('resources')
+
+        candidates = []
+        changed = False
+        resource = self.find_resource(kind="SecurityContextConstraints", api_version="security.openshift.io/v1")
+        for item in sccs:
+            subjects = item.get(key, [])
+            retainedSubjects = [x for x in subjects if x not in ref_names]
+            if len(subjects) != len(retainedSubjects):
+                candidates.append(item['metadata']['name'])
+                if not self.check_mode:
+                    upd_sc = item
+                    upd_sc.update({key: retainedSubjects})
+                    try:
+                        resource.apply(upd_sc, namespace=None)
+                        changed = True
```

Comment:
Same as above. This should be moved down so that check mode correctly reports that a change would happen.

Reply:
Done. Thanks!!